### ISSUE TITLE
loss is NaN for some batches/steps

### ISSUE THREAD

**Issue by neerajwagh** [facebookresearch/barlowtwins#21]
EDIT - After more investigation, the issue is likely with my input data/features. Marking as closed. Sorry about that!

Hi,

Thanks for making it super easy to apply this architecture to other datasets. 

I'm working with spectrogram images containing 10 channels, and a resnet18 backbone. Currently using ADAM optimizer with single-GPU training. My loss computation results in NaNs for some batches (early on, shown below) and then NaNs for the whole epoch (later on, not shown), and I'm not sure why. I was wondering if you had any thoughts on why that might be the case. A snapshot of the output logs (with **on_diag and off_diag printed for each batch/step**) is given below:

**tensor(nan, device='cuda:0', grad_fn=<MulBackward0>) tensor(nan, device='cuda:0', grad_fn=<MulBackward0>)**
tensor(0.0051, device='cuda:0', grad_fn=<MulBackward0>) tensor(124.6055, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0047, device='cuda:0', grad_fn=<MulBackward0>) tensor(116.0811, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0038, device='cuda:0', grad_fn=<MulBackward0>) tensor(116.7065, device='cuda:0', grad_fn=<MulBackward0>)
tensor(nan, device='cuda:0', grad_fn=<MulBackward0>) tensor(nan, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0080, device='cuda:0', grad_fn=<MulBackward0>) tensor(107.8073, device='cuda:0', grad_fn=<MulBackward0>)
**tensor(nan, device='cuda:0', grad_fn=<MulBackward0>) tensor(nan, device='cuda:0', grad_fn=<MulBackward0>)**
tensor(0.0051, device='cuda:0', grad_fn=<MulBackward0>) tensor(115.6731, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0027, device='cuda:0', grad_fn=<MulBackward0>) tensor(132.3802, device='cuda:0', grad_fn=<MulBackward0>)
**tensor(nan, device='cuda:0', grad_fn=<MulBackward0>) tensor(nan, device='cuda:0', grad_fn=<MulBackward0>)**
tensor(0.0864, device='cuda:0', grad_fn=<MulBackward0>) tensor(94.7531, device='cuda:0', grad_fn=<MulBackward0>)
**tensor(nan, device='cuda:0', grad_fn=<MulBackward0>) tensor(nan, device='cuda:0', grad_fn=<MulBackward0>)**
**{"epoch": 1, "batch": 0, "learning_rate": 0.0, "loss": NaN, "time": "0:01:28.300381"}**
tensor(0.0051, device='cuda:0', grad_fn=<MulBackward0>) tensor(116.8903, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0040, device='cuda:0', grad_fn=<MulBackward0>) tensor(117.0997, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0031, device='cuda:0', grad_fn=<MulBackward0>) tensor(128.7704, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0024, device='cuda:0', grad_fn=<MulBackward0>) tensor(120.1153, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0047, device='cuda:0', grad_fn=<MulBackward0>) tensor(112.5160, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0065, device='cuda:0', grad_fn=<MulBackward0>) tensor(116.6054, device='cuda:0', grad_fn=<MulBackward0>)
tensor(0.0031, device='cuda:0', grad_fn=<MulBackward0>) tensor(120.2410, device='cuda:0', grad_fn=<MulBackward0>)
